cmd/podman/trust.go

```diff
@@ -0,0 +1,557 @@
+				}
+			}
+		}
+		if !exists {
+			if policyContentStruct.Transports == nil {
+				policyContentStruct.Transports = make(map[string]RepoMap)
+			}
+			if policyContentStruct.Transports["docker"] == nil {
+				policyContentStruct.Transports["docker"] = make(map[string][]RepoContent)
+			}
+			policyContentStruct.Transports["docker"][args[0]] = append(policyContentStruct.Transports["docker"][args[0]], newReposContent...)
+		}
+	}
+
+	data, err := json.MarshalIndent(policyContentStruct, "", "    ")
+	if err != nil {
+		return errors.Wrapf(err, "error setting trust policy")
+	}
+	err = ioutil.WriteFile(policyPath, data, 0644)
+	if err != nil {
+		return errors.Wrapf(err, "error setting trust policy")
+	}
+	return nil
+}
+
+func getGPGId(keys []string) string {
+	for _, k := range keys {
+		if _, err := os.Stat(k); err != nil {
+			decodeKey, err := base64.StdEncoding.DecodeString(k)
+			if err != nil {
+				logrus.Warnf("error decoding key data")
+				continue
+			}
+			tmpfileName, err := createKeyTmpFile(decodeKey)
+			if err != nil {
+				logrus.Warnf("error creating key date temp file %s", err)
+			}
+			defer os.Remove(tmpfileName)
+			k = tmpfileName
+		}
+		cmd := exec.Command("gpg2", "--with-colons", k)
+		results, err := cmd.Output()
+		if err != nil {
+			logrus.Warnf("error get key identity: %s", err)
+			continue
+		}
+		resultsStr := *(*string)(unsafe.Pointer(&results))
+		scanner := bufio.NewScanner(strings.NewReader(resultsStr))
+		var parseduids []string
+		for scanner.Scan() {
+			line := scanner.Text()
+			if strings.HasPrefix(line, "uid:") || strings.HasPrefix(line, "pub:") {
+				uid := strings.Split(line, ":")[9]
+				if uid == "" {
+					continue
+				}
+				parseduid := uid
+				if strings.Contains(uid, "<") && strings.Contains(uid, ">") {
+					parseduid = strings.SplitN(strings.SplitAfterN(uid, "<", 2)[1], ">", 2)[0]
+				}
+				parseduids = append(parseduids, parseduid)
+			}
+		}
+		return strings.Join(parseduids, ",")
+	}
+	return ""
+}
+
+func createKeyTmpFile(content []byte) (string, error) {
+	tmpfile, err := ioutil.TempFile("/run/", "")
+	if err != nil {
+		return "", err
+	}
+	if _, err := tmpfile.Write(content); err != nil {
+		return "", err
+
+	}
+	if err := tmpfile.Close(); err != nil {
+		return "", err
+	}
+	return tmpfile.Name(), nil
+}
+
+var typeDescription = map[string]string{"insecureAcceptAnything": "accept", "signedBy": "signed", "reject": "reject"}
+
+func trustTypeDescription(t string) string {
+	ele, exist := typeDescription[t]
+	if !exist {
+		logrus.Warnf("invalid trust type %s", t)
+	}
+	return ele
+}
+
+func sortPolicyJSONKey(m map[string]map[string]interface{}) []string {
+	keys := make([]string, len(m))
+	i := 0
+	for k := range m {
+		keys[i] = k
+		i++
+	}
+	sort.Strings(keys)
+	return keys
+}
+
+// defaultPolicyPath returns a path to the default policy of the system.
+func defaultPolicyPath(sys *types.SystemContext) string {
+	systemDefaultPolicyPath := "/etc/containers/policy.json"
```

Comment:
We should get this from containers/image, not hard code it here.

Reply:
Shouldn't we be using the values `libpod/image` set? If we're working with image trust, we should be mirroring how `libpod` will actually see the images

Reply:
how to use that? the value from `image.Runtime.SignaturePolicyPath`?

Reply:
Yeah, that would be it